rclcpp/src/rclcpp/executors/static_executor_waitable.cpp

```diff
@@ -0,0 +1,264 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rclcpp/executors/static_single_threaded_executor.hpp"
+#include "rclcpp/executors/static_executor_waitable.hpp"
+
+using rclcpp::executors::StaticExecutorWaitable;
+
+StaticExecutorWaitable::~StaticExecutorWaitable()
+{
+  // Disassociate all nodes
+  for (auto & weak_node : weak_nodes_) {
+    auto node = weak_node.lock();
+    if (node) {
+      std::atomic_bool & has_executor = node->get_associated_with_executor_atomic();
+      has_executor.store(false);
+    }
+  }
+  exec_list_.clear();
+  weak_nodes_.clear();
+  guard_conditions_.clear();
+}
+
+void
+StaticExecutorWaitable::init(
+       rcl_wait_set_t* p_wait_set,
+       memory_strategy::MemoryStrategy::SharedPtr& memory_strategy,
+       StaticExecutorWaitable::SharedPtr this_shared_ptr)
+{
+  // Empty initialize executable list
+  executor::ExecutableList exec_list_();
```

Comment:
This line looks wrong...

I'm not sure if it's needed, but in case it should be
```
exec_list = executor::ExecutableList();
```

Reply:
In your case, I'm doing an extra copy. 
What I'm doing is just creating an object of type `executor::ExecutableList` called `exec_list_` and calling the default constructor `()`. 
But what I'm not sure if the compiler knows that this `exec_list_` is the same as the member variable. 
What I can do to be sure, is delete this line, and in the declaration of member variables call directly the default initialization: `executor::ExecutableList exec_list_{};`

Reply:
You are creating a new variable, it's not what you should do.

Do not care about the extra copy, doing it in the init function allows the object to be re-initialized if needed